include/wil/common.h

```diff
@@ -83,6 +83,12 @@
 #endif
 /// @endcond
 
+/// @cond
+#if defined(__MINGW32__) && !defined(_NATIVE_WCHAR_T_DEFINED)
+#define __wchar_t wchar_t

```

Comment:
Maybe better as a `typedef`?

Reply:
Or are there other places that define this macro that it may otherwise conflict with?

Reply:
Yes, that's better, done

Reply:
Now that I think about it, maybe it should be:
```
#if !defined(_MSC_VER) && !defined(_NATIVE_WCHAR_T_DEFINED)
```
mingw is about include headers, and this is a compiler feature, right?

Reply:
According to [MDSN](https://learn.microsoft.com/en-us/cpp/build/reference/zc-wchar-t-wchar-t-is-native-type?view=msvc-170), `_NATIVE_WCHAR_T_DEFINED` is more about whether `wchar_t` maps to `unsigned char` or `__wchar_t`. So I actually don't think checking for it in this macro is correct. Idk how to correctly handle this with other compilers; it might be that `_MSC_VER` is the proper check, but idk since I've already been proven wrong once today on that

Reply:
OK, well, I know that both options work for clang which is what I need. I assume it'd be the same with gcc. So I don't mind having it merged in any form. That was just a pedantic thought. We can change it later if needed.